test/iht/views/application/SubmittableApplicationPageBehaviour.scala

```diff
@@ -46,7 +48,11 @@ trait SubmittableApplicationPageBehaviour[A] extends ApplicationPageBehaviour {
       continueLink.attr("value") shouldBe messagesApi("iht.continue")
 
       val cancelLink = view.getElementById("cancel-button")
-      cancelLink.attr("href") shouldBe cancelUrl.url
+      if(linkHash > ""){
```

Comment:
Better, clearer, to use .nonEmpty here. Personally I'd probably write it like this:
val expectedResult = if(linkHash.nonEmpty) attrib.target.url + linkHash else attrib.target.url
cancelButton.attr("href") shouldBe expectedResult
but that's perhaps more a matter of personal taste.
Also, this bit of code is repeated 4 times throughout tests, so I'd put it somewhere common such as TestHelper.